### ISSUE TITLE
incompatibility: wrong nested beforeEach execution order

### ISSUE THREAD

**Issue by mpareja** [nicolo-ribaudo/jest-light-runner#39]
The following test passes with jest but not jest-light-runner. The `beforeAll` behaviour appears to be correct, though.

```javascript
describe('beforeEach execution order', () => {
  const calls = []
  beforeEach(() => {
    calls.push('outer')
  })

  describe('nested', () => {
    beforeEach(() => {
      calls.push('inner')
    })

    it('expect outer beforeEach before inner', () => {
      expect(calls).toEqual(['outer', 'inner'])
    })
  })
})
```

**Comment by nicolo-ribaudo** [comment]
Thanks! I'll fix this in the next few days, unless you want to open a pull request 🙂

**Comment by nicolo-ribaudo** [comment]
Fixed by #44
